playbook/actions/pod.go

```diff
@@ -0,0 +1,116 @@
+package actions
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/flanksource/commons/logger"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	"github.com/flanksource/incident-commander/api"
+	v1 "github.com/flanksource/incident-commander/api/v1"
+)
+
+const (
+	defaultMaxLength        = 3000
+	defaultContainerTimeout = 5 * time.Minute
+
+	sidecarContainerName     = "container-waiter"
+	containerImage           = "ubuntu:jammy"
+	playbookActionSelector   = "playbooks.mission-control.flanksource.com/actions"
+	playbookActionLabelValue = "playbooks-pod-action"
+)
+
+type PodResult struct {
+	Logs string
+}
+
+type Pod struct {
+}
+
+func (c *Pod) Run(ctx api.Context, action v1.PodAction, env TemplateEnv) (*PodResult, error) {
+	timeout := time.Duration(action.Timeout) * time.Minute
+	if timeout == 0 {
+		timeout = defaultContainerTimeout
+	}
+
+	if action.MaxLength <= 0 {
+		action.MaxLength = defaultMaxLength
```

Comment:
I don't think we should trim by default - its good to add an option, but not as a default - Long term we would store there logs in S3 like store